### DISCUSSION TITLE
Lookbook v1.0 🎉

### DISCUSSION THREAD

**Discussion by allmarkedup** [lookbook-hq/lookbook#129] [Announcements]
**Lookbook v1.0** has just been released with a bunch of new features and a completely rebuilt ViewComponent-based UI which should make new feature development quicker and easier.

Check out the new [Lookbook documentation site](https://lookbook.build) for details on upgrading and documentation on all the new features.

Got questions or comments about the v1.0.0 release? Post them below or [open an issue](https://github.com/allmarkedup/lookbook/issues).

---

### Frontend changes:

* Completely re-written UI - now built with ViewComponents
* New [UI themes and customisation options](https://lookbook.build/guide/themes/)
* Improved small-screen/mobile view
* Debug menu
* ...and many other small visual and usability improvements

### New options for extending Lookbook:

* Define [custom tags](https://lookbook.build/guide/extend/tags/) for preview file annotations
* Create [your own tab panels](https://lookbook.build/guide/extend/panels/) - with complete control over content/layout and access to all Lookbook data
* Use [lifecycle hooks](https://lookbook.build/guide/extend/hooks/) to run your own code when Lookbook starts up, when files change or at shutdown

### Other changes

* 'Workbench' app for developing Lookbook's UI components in Lookbook
* New Rspec-based test suite and dummy app
* All new [documentation site](https://lookbook.build) built using [Bridgetown](https://www.bridgetownrb.com/)

**Comment by paulrobertlloyd** [comment]
Congratulations, and well done hitting this important milestone!

**Reply by allmarkedup** [reply]
Thanks @paulrobertlloyd! And hope you are well mate.

1.0 is just a number really but I feel it's a nice way to indicate that the thing you are building is at a stage where people can expect to rely on it. Getting the extension APIs in there and cleaning up the frontend was really important for me to get done before I could think about tagging it v1.0. 

Are you using Lookbook currently in any capacity? If so I'd always value any comments/advice/suggestions you might have, as always :-)

**Reply by paulrobertlloyd** [reply]
I congratulate you on reaching v1.0 partially with a degree of envy as I’m attempting to do the same with [my own side project](https://github.com/getindiekit/indiekit).

I’m not using Lookbook, but have been using ViewComponents extensively in various Rails-based government services, and just generally like to be nosey, you know me :wink: There was at one point an attempt to use this project for the [GOV.UK Components](https://github.com/DFE-Digital/govuk-components) gem, but in the end that project took a different route for reasons I forget.

**Reply by allmarkedup** [reply]
@paulrobertlloyd indiekit looks great! Nice work mate. Looks like *a lot* of work though, respect for pulling that all together.

Seems like you are chipping away at that 1.0 release as well!

**Comment by DRBragg** [comment]
Congrats @allmarkedup! This is awesome!

The new docs site looks great and I'm excited to try out the new custom tags feature 😁

**Reply by allmarkedup** [reply]
@DRBragg awesome - let me know how you get on! Excited to see what people start to use it for 🤓
